dev-tools/globals.ts

```diff
@@ -155,176 +155,490 @@ interface SecondaryEffect {
 	self?: SelfEffect
 	status?: string
 	volatileStatus?: string
-	onAfterHit?: EffectData["onAfterHit"]
-	onHit?: EffectData["onHit"]
+	onAfterHit?: MoveEventMethods["onAfterHit"]
+	onHit?: MoveEventMethods["onHit"]
 }
 
-interface EventMethods {
+interface CommonHandlers {
+	ModifierEffect: (this: Battle, relayVar: number, target: Pokemon, source: Pokemon, effect: Effect) => number | void;
+	ModifierMove: (this: Battle, relayVar: number, target: Pokemon, source: Pokemon, move: ActiveMove) => number | void;
+	ResultMove: boolean | ((this: Battle, target: Pokemon, source: Pokemon, move: ActiveMove) => boolean | null | "" | void);
+	ExtResultMove: boolean | ((this: Battle, target: Pokemon, source: Pokemon, move: ActiveMove) => boolean | null | number | "" | void);
+	VoidEffect: (this: Battle, target: Pokemon, source: Pokemon, effect: Effect) => void;
+	VoidMove: (this: Battle, target: Pokemon, source: Pokemon, move: ActiveMove) => void;
+	ModifierSourceEffect: (this: Battle, relayVar: number, source: Pokemon, target: Pokemon, effect: Effect) => number | void;
+	ModifierSourceMove: (this: Battle, relayVar: number, source: Pokemon, target: Pokemon, move: ActiveMove) => number | void;
+	ResultSourceMove: boolean | ((this: Battle, source: Pokemon, target: Pokemon, move: ActiveMove) => boolean | null | "" | void);
+	ExtResultSourceMove: boolean | ((this: Battle, source: Pokemon, target: Pokemon, move: ActiveMove) => boolean | null | number | "" | void);
+	VoidSourceEffect: (this: Battle, source: Pokemon, target: Pokemon, effect: Effect) => void;
+	VoidSourceMove: (this: Battle, source: Pokemon, target: Pokemon, move: ActiveMove) => void;
+}
+
+interface AbilityEventMethods {
+	onCheckShow?: (this: Battle, pokemon: Pokemon) => void
+	onEnd?: (this: Battle, target: Pokemon & Side & Field) => void
+	onPreStart?: (this: Battle, pokemon: Pokemon) => void
+	onStart?: (this: Battle, target: Pokemon) => void
+}
+
+interface ItemEventMethods {
+	onEat?: ((this: Battle, pokemon: Pokemon) => void) | false
+	onPrimal?: (this: Battle, pokemon: Pokemon) => void
+	onStart?: (this: Battle, target: Pokemon) => void
+	onTakeItem?: ((this: Battle, item: Item, pokemon: Pokemon, source: Pokemon, move?: ActiveMove) => boolean | void) | boolean
+}
+
+interface MoveEventMethods {
 	/** Return true to stop the move from being used */
 	beforeMoveCallback?: (this: Battle, pokemon: Pokemon, target: Pokemon | null, move: ActiveMove) => boolean | void
 	beforeTurnCallback?: (this: Battle, pokemon: Pokemon, target: Pokemon) => void
 	damageCallback?: (this: Battle, pokemon: Pokemon, target: Pokemon) => number | false
+
+	onAfterHit?: CommonHandlers["VoidSourceMove"]
```

Comment:
does Typescript not like `'`? I dont know by our string rules that `"` is approriate?

Reply:
This file was already using double quotes, but originally it should use single quotes, yes.

Reply:
Or, better-yet, dot-notation